File: output/go_graphql_sdk/client_sdk/models/graph_qlrequest.go
package 

import (
	"encoding/json"
	"time"
)

// Graphqlrequest GraphQL request object
type Graphqlrequest struct {
	// GraphQL query string
Query String `json:"query"`
}

// NewGraphqlrequest creates a new Graphqlrequest with required fields
func NewGraphqlrequest(query String) *Graphqlrequest {
	return &Graphqlrequest{
		Query: query,
	}
}



// Validate validates the Graphqlrequest instance
func (m *Graphqlrequest) Validate() error {
	// Add validation logic here if needed
	// Query is required
	return nil
}

// String returns a string representation of the Graphqlrequest
func (m *Graphqlrequest) String() string {
	data, _ := json.MarshalIndent(m, "", "  ")
	return string(data)
}

File: output/rust_graphql_sdk/src/models/graph_qlrequest.rs
//! GraphQL request object
//! 

use serde::{Deserialize, Serialize};
use std::collections::HashMap;
use chrono::{DateTime, NaiveDate, Utc};

/// GraphQL request object
#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
pub struct Graphqlrequest {
    /// GraphQL query string
    #[serde(rename = "query")]
    pub query: String,
    
}

impl Graphqlrequest {
    /// Create a new instance with required fields
    pub fn new(query: String) -> Self {
        Self {
            query,
        }
    }
    
    
    
    
    /// Validate the instance
    pub fn validate(&self) -> Result<(), String> {
        // Add validation logic here if needed
        // query is required and present
        Ok(())
    }
}

impl Default for Graphqlrequest {
    fn default() -> Self {
        Self {
            query: Default::default(),
        }
    }
}

impl std::fmt::Display for Graphqlrequest {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        write!(f, "Graphqlrequest \{ ")?;
        write!(f, "query: {:?}", self.query)?;
        write!(f, " \}")
    }
}